sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -178,29 +178,21 @@ public void onResponse(okhttp3.Call call, okhttp3.Response response) {
     private static class OkHttpResponse extends HttpResponse {
         private final int statusCode;
         private final HttpHeaders headers;
-        private final Mono<ResponseBody> responseBodyMono;
+        private final ResponseBody responseBody;
         // using 4K as default buffer size: https://stackoverflow.com/a/237495/1473510
         private static final int BYTE_BUFFER_CHUNK_SIZE = 4096;
 
         OkHttpResponse(Response innerResponse, HttpRequest request) {
             super(request);
             this.statusCode = innerResponse.code();
             this.headers = fromOkHttpHeaders(innerResponse.headers());
-            if (innerResponse.body() == null) {
-                // innerResponse.body() getter will not return null for server returned responses.
-                // It can be null:
-                // [a]. if response is built manually with null body (e.g for mocking)
-                // [b]. for the cases described here
-                // [ref](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response/body/).
-                //
-                this.responseBodyMono = Mono.empty();
-            } else {
-                this.responseBodyMono = Mono.using(innerResponse::body,
-                    Mono::just,
-                    // Resource cleanup
-                    // square.github.io/okhttp/4.x/okhttp/okhttp3/-response-body/#the-response-body-must-be-closed
-                    ResponseBody::close, /* Change in behavior since reactor-core 3.3.0.RELEASE */ false);
-            }
+            // innerResponse.body() getter will not return null for server returned responses.
+            // It can be null:
+            // [a]. if response is built manually with null body (e.g for mocking)
+            // [b]. for the cases described here
+            // [ref](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response/body/).
+            //
+            this.responseBody = innerResponse.body();
```

Comment:
Do we still need the null check for the above mocking (a) and cacheResponse, networkResponse etc (b)?

Reply:
we do that in each of the places `this.responseBody` is used.